packages/salesforcedx-webview-ui/README.md

```diff
@@ -0,0 +1,187 @@
+|   |                      +-----------------+  |    |                          |   +-----------------+   |
+|   |                      |                 |  |    |      +-----------+       |   |                 |   |
+|   |                      |                 |  |    |      |postMessage|       |   |                 |   |
+|   |                      | +-------------+ |  |  --+------+-----------+-------+>  | +-------------+ |   |
+|   |                      | |EventListener| |  |    |                          |   | |EventListener| |   |
+|   |                      | +-------------+ |  |    |                          |   | +-------------+ |   |
+|   |                      |                 |  |    |                          |   |                 |   |
+|   |                      |                 |  |    |                          |   |                 |   |
+|   |                             <----------+  |  <-+------+-----------+-------+-         <----------+   |
+|   |                                           |    |      |postMessage|       |                         |
+|   +-------------------------------------------+    |      +-----------+       |                         |
+|                    HTML Page #2                    |                          +-------------------------+
+|                                                    |
++----------------------------------------------------+                               VS Code Extension
+                   WebviewPanel #2
+
+Created with Monodraw
+```
+
+We can embed an HTML page that contains our components into a WebviewPanel.
+Those components can communicate with the VS Code Extension via message passing.
+Similarly, the VS Code Extension can communicate with the WebviewPanel via
+message passing. The communication protocol is up to the implementor. 
+
+There is no limit on the number of WebviewPanels that we can embed. However,
+since each one WebviewPanel is backed by an Electron WebView, which runs in its
+own process, there could be performance penalties.
+
+# Third-party Libraries
+
+* [react-script-ts](https://github.com/wmonk/create-react-app-typescript) - Provides the template for the project structure
+* [create-react-app](https://github.com/facebook/create-react-app) - The basis for the project structure. Read this to understand how things work underneath.
+* [Blueprint](http://blueprintjs.com/docs/v2/) - For UI components
+* [Jest](https://facebook.github.io/jest/) - For testing
+
+# Useful Reading
+
+* [`<webview>` Tag](https://electronjs.org/docs/api/webview-tag)
+* [Interop’s Labyrinth: Sharing Code Between Web & Electron Apps](https://slack.engineering/interops-labyrinth-sharing-code-between-web-electron-apps-f9474d62eccc)
+* [Growing Pains: Migrating Slack’s Desktop App to BrowserView](https://slack.engineering/growing-pains-migrating-slacks-desktop-app-to-browserview-2759690d9c7b)
```

Comment:
There are some bugs with webview in Electron. See https://github.com/electron/electron/issues?q=is%3Aopen+is%3Aissue+label%3Acomponent%2Fwebview

However, I searched on the VS Code repo and there were [no](https://github.com/Microsoft/vscode/issues?utf8=%E2%9C%93&q=browserview) discussions about considering using `BrowserView` instead of `WebView`. So that option doesn't seem to exist for VS Code.